os/pm/pm_process.c

```diff
@@ -0,0 +1,153 @@
+
+/************************************************************************
+ * Pre-processor Definitions
+ ************************************************************************/
+
+/************************************************************************
+ * Private Type Declarations
+ ************************************************************************/
+
+/************************************************************************
+ * Private Variables
+ ************************************************************************/
+
+/************************************************************************
+ * Private Functions
+ ************************************************************************/
+
+static uint32_t pm_gettimeinterval(void)
+{
+	uint32_t ret = 0;
+#if defined(CONFIG_PM_LCD_NORMAL_TO_IDLE_THRESH) && defined(CONFIG_PM_LCD_IDLE_TO_STANDBY_THRESH) && defined(CONFIG_PM_STANDBY_TO_SLEEP_THRESH)
+	if (g_pmglobals.state == PM_NORMAL) {
+		ret = CONFIG_PM_LCD_NORMAL_TO_IDLE_THRESH;
+	} else if (g_pmglobals.state == PM_IDLE) {
+		ret = CONFIG_PM_LCD_IDLE_TO_STANDBY_THRESH;
+	} else if (g_pmglobals.state == PM_STANDBY) {
+		ret = CONFIG_PM_STANDBY_TO_SLEEP_THRESH;
+	}
+	DEBUGASSERT(ret != 0);
+#endif
+	return ret;
+}
+
+static void pm_process(int argc)
+{
+	enum pm_state_e newstate;
+	newstate = pm_checkstate();
+	if (newstate > g_pmglobals.state) {
+		/* After timer expiration change PM State to next feasible state */
+		(void)pm_changestate(newstate);
+		/* Resume Timer for power saving */
+		if ((g_pmglobals.state != PM_SLEEP)) {
+			(void)pm_process_resume();
+		}	
+	}
+}
+
+/************************************************************************
+ * Public Functions
+ ************************************************************************/
+
+/****************************************************************************
+ * Name: pm_process_init
+ *
+ * Description:
+ *   It initializes the pm state transition timer.
+ *
+ * Input Parameters:
+ *
+ * Returned Value:
+ *   0 (OK), if able to start timer
+ * 	 -1 (ERROR), on error.
+ *
+ ****************************************************************************/
+
+int pm_process_init(void)
+{
+	/* Create WDOG timer if it is not already there */
+	if (!g_pmglobals.wdog) {
+		g_pmglobals.wdog = wd_create();
+		if (g_pmglobals.wdog == NULL) {
+			pmvdbg("Unable to create WDog timer\n");
+			return ERROR;
+		}
+	}
+	return OK;
+}
+
+/****************************************************************************
+ * Name: pm_process_suspend
+ *
+ * Description:
+ *   It stops the pm state transition timer.
+ *
+ * Input Parameters:
+ *
+ * Returned Value:
+ *   0 (OK), if able to start timer
+ * 	 -1 (ERROR), on error.
+ *
+ ****************************************************************************/
+
+int pm_process_suspend(void)
+{
+	enum pm_state_e newstate;
+	newstate = pm_checkstate();
+	if (newstate < g_pmglobals.state) {
+		/* Change to high power state immediately */
```

Comment:
Is there any possibility of returning error in this function?

Reply:
Currently pm_checkstate can return ERROR  but after merging PR https://github.com/Samsung/TizenRT/pull/6259, there will be no possibility.

Reply:
If that is the case then this API will always return OK then why do we need int return type. we can very well have void.